### PR TITLE
docs to hold maintainer build platforms

### PR DESCRIPTION
**Pull request by deitch**
Signed-off-by: Avi Deitcher <avi@deitcher.net>

**- What I did**

Created a doc to track how maintainers can manage their build platforms. Also includes directions for getting modern Docker on s390x.

**- How I did it**

Markdown.

**- How to verify it**

Read it

**- Description for the changelog**

Maintainer build server logs

### CODE REVIEW

**Comment by rn** [comment]
Thanks for documenting, I had  gist somewhere for the steps... It might better to frame this not just for maintainers but more like: "Here is how you set up your LinuxKit development machine"

I can't remember how I installed docker on s390x but it was certainly different to what you describe. But whatever works.

To run tests you need something like `apt-get install -y jq expect qemu-kvm`

I would also mention that `manifest-tool` and `rtf`need to be installed (`make bin/manifest-tool` `make bin/rtf` and then have it somewhere in your `$PATH`). 

finally I had:
```
usermod -aG docker $USER
usermod -aG kvm $USER
usermod -aG sudo $USER
```
to allow a normal user to run docker/qemu

File: docs/maintainer-build-platforms
Comment (rn) [inline docs/maintainer-build-platforms:8]: I think Docker recommends using the docker install from their channels rather than using the distro packages. I think it's better since you have better control over which version of docker you are using. https://docs.docker.com/engine/install/

Comment (deitch) [reply docs/maintainer-build-platforms:8]: Yeah, I was implying that, but wasn't too clear. I really just wanted to capture the messy s390x steps before I forgot. I will update it.

**Comment by deitch** [comment]
Updated with everything you suggested @rn

**Review by dave-tucker — DISMISSED** [review]
LGTM

**Comment by deitch** [comment]
Thanks @dave-tucker . Going to await CI, and @rn signoff after I addressed his comments, and will merge in.

File: docs/maintainer-build-platforms.md
Comment (thaJeztah) [inline docs/maintainer-build-platforms.md:20]: Updated packages should be available soon (it's being worked on with IBM)

File: docs/maintainer-build-platforms.md
Comment (thaJeztah) [inline docs/maintainer-build-platforms.md:27]: I should note here that currently RHEL is only published for s390x; for other arches, the CentOS 7/8 packages should work

File: docs/maintainer-build-platforms.md
Comment (deitch) [reply docs/maintainer-build-platforms.md:20]: You mean all of that work was for nothing? :-)

Not really; it got me going today. Let's keep this as is, and we can update it when the updated packages are ready.

File: docs/maintainer-build-platforms.md
Comment (deitch) [reply docs/maintainer-build-platforms.md:27]: Good point.

File: docs/maintainer-build-platforms.md
Comment (rn) [inline docs/maintainer-build-platforms.md:1]: I would rename the file to `developer-setup.md` or similar. It shouldn't be maintainer specific. Here is how you setup you s390x system (you have in your garage) for doing LinuxKit development

**Review by rn — COMMENTED** [review]
really just a nit on the filename, but otherwise LGTM

File: docs/maintainer-build-platforms.md
Comment (thaJeztah) [reply docs/maintainer-build-platforms.md:20]: (hides) _hopefully_ soon the work becomes redundant yes. Until then 👍 👍  (thanks!). Your PR just caught my eye, and I thought I'd give you a heads-up.

Comment (deitch) [reply docs/maintainer-build-platforms.md:20]: Definitely happy for it!

File: docs/maintainer-build-platforms.md
Comment (deitch) [reply docs/maintainer-build-platforms.md:1]: I like that. Will do.

File: docs/maintainer-build-platforms.md
Comment (deitch) [reply docs/maintainer-build-platforms.md:27]: Will mark it there.

**Comment by deitch** [comment]
Updated everything per the comments from @rn and @thaJeztah . Looking for one of them to give a "good to go", and we can merge in.

**Review by rn — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

If you want to try s390x packages; some packages for Ubuntu are currently up on staging (but we're still updating some parts of our release with IBM's build pipeline); https://download-stage.docker.com/linux/ubuntu/dists/focal/pool/stable/s390x/

**Pull request merged**
